cmd/regex_compare.go

```diff
@@ -159,24 +180,39 @@ func readCurrentRegex(filePath string, ruleId string, chainOffset uint8) string
 	idRegex := regexp.MustCompile(fmt.Sprintf("id:%s", ruleId))
 	index := 0
 	var line []byte
+	foundRule := false
+	chainCount := uint8(0)
 	for index, line = range lines {
-		if idRegex.Match(line) {
+		if !foundRule && idRegex.Match(line) {
+			foundRule = true
+			if chainOffset == 0 {
+				index--
+				break
+			}
+			continue
+		}
+		if foundRule && secRuleRegex.Match(line) {
+			chainCount++
+		}
+		if foundRule && chainCount == chainOffset {
 			break
 		}
 	}
-	regexLine := lines[index-1]
-	regexRegex := regexp.MustCompile(`.*"@rx (.*)" \\`)
-	found := regexRegex.FindAllStringSubmatch(string(regexLine), -1)
+	if !foundRule || chainOffset != chainCount {
+		logger.Fatal().Msgf("Failed to find rule %s, chain offset, %d in %s", ruleId, chainOffset, filePath)
+	}
+	regexLine := lines[index]
+	found := ruleRxRegex.FindAllStringSubmatch(string(regexLine), -1)
```

Comment:
Ugh, needed to dig other files to understand this was defined [here](https://github.com/theseion/crs-toolchain/blob/f8d5a7cefbb3a01de0994ac8ba1a05adbccd6913/cmd/regex_update.go#L26). Maybe we want to move all regex definitions to a common file?

Reply:
Yes, that might be the best thing to do. I hate it that "private" variables and functions have package scope.

Reply:
Done